.github/workflows/docs.yml

```diff
@@ -0,0 +1,47 @@
+name: Build and Deploy API Documentation
+
+on:
+  push:
+    tags:
+      # Match 1.0.0 but not 1.0.0-rc1 or other suffixes
+      - '[0-9]+\.[0-9]+\.[0-9]'
+      - '![0-9]+\.[0-9]+\.[0-9]-[a-z]*'
+      - '![0-9]+\.[0-9]+\.[0-9]-[A-Z]*'
```

Comment:
❓ Why do we need the following?

```
- '![0-9]+\.[0-9]+\.[0-9]-[a-z]*'
- '![0-9]+\.[0-9]+\.[0-9]-[A-Z]*'
```

💡 Semantic versioning website has some [suggested regex](https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string). Those validate all the possibilities of SemVer, so we can't apply them directly. However, I like the fact that they validate errors like: `01.09.01`. However, after checking the documentation, I'm not sure if the `or` is available on GitHub.

What we can do, at least, is limit mayor to 1>=, as we know we won't release anything as (`0.X.X`). I think, we also need to allow patch version 9>, something like:

`'[1-9]+\.[0-9]+\.[0-9]+'`

Reply:
>❓ Why do we need the following?

These are to prevent running that workflow on anything like: `1.1.1-rc1` or `1.1.1-ALPHA1`. We removed docs for versions like that recently so I thought we don't want to have docs for those.

>'[1-9]+\.[0-9]+\.[0-9]+'

Sounds good to me. I wasn't too worried about using 0 as a major. Technically this is correct, and if that error would happen on our side we could just remove the last commit. But yeah, at this point, 0 as major, shouldn't be used.

Reply:
But `1.1.1-rc1` or `1.1.1-ALPHA1` don't match with the regex `[1-9]+.[0-9]+.[0-9]+`, right?

Reply:
They would I think. This `^[1-9]+.[0-9]+.[0-9]+` with the`^` would match the beginning, but like I said, I'm not sure we can use all the characters as they are not mentioned in the docs.

Reply:
If they look for a full match in the tag (which makes sense to me), it shouldn't work:

<img width="253" alt="image" src="https://github.com/user-attachments/assets/a52efc47-056c-4982-8f92-14b43eeeab4d">

Anyway, it won't hurt those lines, so feel free to keep them.

Reply:
Right, so I did test it by pushing some tags and indeed they do a full match. I have updated the workflow to use `[1-9]+.[0-9]+.[0-9]+`